File: chain-sdk/javascript-closure-angular/API/Client/UniswapDayDataV3DTO.js
goog.provide('API.Client.UniswapDayDataV3DTO');

/**
 * Data accumulated and condensed into day stats for all of Uniswap
 * @record
 */
API.Client.UniswapDayDataV3DTO = function() {}

/**
 * @type {!Date}
 * @export
 */
API.Client.UniswapDayDataV3DTO.prototype.entryTime;

/**
 * @type {!Date}
 * @export
 */
API.Client.UniswapDayDataV3DTO.prototype.recvTime;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.UniswapDayDataV3DTO.prototype.blockNumber;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.UniswapDayDataV3DTO.prototype.vid;

/**
 * timestamp rounded to current day by dividing by 86400
 * @type {!string}
 * @export
 */
API.Client.UniswapDayDataV3DTO.prototype.id;

/**
 * timestamp rounded to current day by dividing by 86400
 * @type {!number}
 * @export
 */
API.Client.UniswapDayDataV3DTO.prototype.date;

/**
 * total volume across all pairs on this day, stored as a derived amount of ETH
 * @type {!string}
 * @export
 */
API.Client.UniswapDayDataV3DTO.prototype.volumeEth;

/**
 * total volume across all pairs on this day, stored as a derived amount of USD
 * @type {!string}
 * @export
 */
API.Client.UniswapDayDataV3DTO.prototype.volumeUsd;

/**
 * total daily volume in Uniswap derived in terms of USD untracked
 * @type {!string}
 * @export
 */
API.Client.UniswapDayDataV3DTO.prototype.volumeUsdUntracked;

/**
 * fees in USD
 * @type {!string}
 * @export
 */
API.Client.UniswapDayDataV3DTO.prototype.feesUsd;

/**
 * @type {!API.Client.BigInteger}
 * @export
 */
API.Client.UniswapDayDataV3DTO.prototype.txCount;

/**
 * tvl in terms of USD
 * @type {!string}
 * @export
 */
API.Client.UniswapDayDataV3DTO.prototype.tvlUsd;

File: chain-sdk/clojure/src/on_chain_api/specs/uniswap_day_data_v3_dto.clj
(ns on-chain-api.specs.uniswap-day-data-v3-dto
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [on-chain-api.specs.big-integer :refer :all]
            )
  (:import (java.io File)))


(def uniswap-day-data-v3-dto-data
  {
   (ds/opt :entry_time) inst?
   (ds/opt :recv_time) inst?
   (ds/opt :block_number) int?
   (ds/opt :vid) int?
   (ds/opt :id) string?
   (ds/opt :date) int?
   (ds/opt :volume_eth) string?
   (ds/opt :volume_usd) string?
   (ds/opt :volume_usd_untracked) string?
   (ds/opt :fees_usd) string?
   (ds/opt :tx_count) big-integer-spec
   (ds/opt :tvl_usd) string?
   })

(def uniswap-day-data-v3-dto-spec
  (ds/spec
    {:name ::uniswap-day-data-v3-dto
     :spec uniswap-day-data-v3-dto-data}))
